app/Model/EventReport.php

```diff
@@ -410,71 +406,86 @@ public function reArrangeReport(array $report)
     /**
      * getProxyMISPElements Extract MISP Elements from an event and make them accessible by their UUID
      *
-     * @param  array $user
-     * @param  int|string $eventid
+     * @param array $user
+     * @param int|string $eventid
      * @return array
+     * @throws Exception
      */
     public function getProxyMISPElements(array $user, $eventid)
     {
-        $event = $this->Event->fetchEvent($user, ['eventid' => $eventid]);
+        $options = [
+            'noSightings' => true,
+            'sgReferenceOnly' => true,
+            'noEventReports' => true,
+            'noShadowAttributes' => true,
+        ];
+
+        $event = $this->Event->fetchEvent($user, array_merge(['eventid' => $eventid], $options));
         if (empty($event)) {
             throw new NotFoundException(__('Invalid Event'));
         }
         $event = $event[0];
-        $parentEventId = $this->Event->fetchSimpleEventIds($user, ['conditions' => [
-            'uuid' => $event['Event']['extends_uuid']
-        ]]);
-        $completeEvent = $event;
-        if (!empty($parentEventId)) {
-            $parentEvent = $this->Event->fetchEvent($user, ['eventid' => $parentEventId, 'extended' => true]);
+
+        if (!empty($event['Event']['extends_uuid'])) {
+            $parentEvent = $this->Event->fetchEvent($user, array_merge(['event_uuid' => $event['Event']['extends_uuid']], $options));
             if (!empty($parentEvent)) {
-                $parentEvent = $parentEvent[0];
-                $completeEvent = $parentEvent;
+                $event = $parentEvent[0];
             }
         }
-        $attributes = Hash::combine($completeEvent, 'Attribute.{n}.uuid', 'Attribute.{n}');
-        $this->AttributeTag = ClassRegistry::init('AttributeTag');
+
```

Comment:
We should record event tags from both the parent and extending event

Reply:
Can you please describe, how it should works? Image that I have three events:
```
Parent event A
|- Child event B
|- Child event C
```
When I am editing report for *Child event B*, should I use attributes, objects and tags from all three events or just from *Child event B* and * Parent event A*?

Reply:
- My idea is to use Attributes of `event A` (if the extended flag is provided, it will includes attributes from its childs)
- Same goes for objects
- For tags, I thought it would make sense to use *event tags* from `A` and `B` and not `C` - but it's debatable

Reply:
OK, so I change the implementation to follow your rules. I can change it any time.